tensorflow_io/avro/20190307-avro-dataset.md

```diff
@@ -0,0 +1,299 @@
+
+### API Design
+
+To showcase the API we define an example record of a person that has a name, address, friends, jobs and coworkers at jobs, and cars, where a car is defined through its color and engine (see Listing 1). Note, that we formally defined the record recursively to keep the example short--however, recursive type definitions are not supported by the avro dataset.
+
+```json
+{  
+   "doc":"Person with friends and coworkers",
+   "namespace":"com.test",
+   "type":"record",
+   "name":"person",
+   "fields":[  
+      {  
+         "name":"name",
+         "type":"record",
+         "fields":[  
+            {  
+               "name":"first",
+               "type":"string"
+            },
+            {  
+               "name":"initial",
+               "type":"string"
+            },
+            {  
+               "name":"second",
+               "type":"string"
+            }
+         ]
+      },
+      {  
+         "name":"address",
+         "type":"record",
+         "fields":[  
+            {  
+               "name":"street",
+               "type":"string"
+            }
+         ]
+      },
+      {  
+         "name":"gender",
+         "type":"string"
+      },
+      {  
+         "name":"friends",
+         "type":"array",
+         "items":"com.test.person"
+      },
+      {  
+         "name":"jobs",
+         "type":"array",
+         "items":{  
+            "name":"coworkers",
+            "type":"array",
+            "items":"com.test.person"
+         }
+      },
+      {  
+         "name":"car",
+         "type":"record",
+         "fields":[  
+            {  
+               "name":"color",
+               "type":"string"
+            },
+            {  
+               "name":"engine",
+               "type":"record",
+               "fields":[  
+                  {  
+                     "name":"id",
+                     "type":"int"
+                  },
+                  {  
+                     "name":"power",
+                     "type":"float"
+                  },
+                  {  
+                     "name":"cylinders",
+                     "type":"int"
+                  },
+                  {  
+                     "name":"sparkplugs",
+                     "type":"int"
+                  }
+               ]
+            }
+         ]
+      },
+      {  
+         "name":"cars",
+         "type":"map",
+         "values":"com.test.car"
+      }
+   ]
+}
+```
+**Listing 1** defines a person with name, address, friends, jobs, and cars.  Note, that the definition of the nested person would require a union with a nested type to be valid--but to simplify we left this out here.
+
+We start of the discussion of the API through examples (see Table 2 and Table 3). These examples list the key expressions together with an explanation. The status indicates whether this is a new, or the same as in the existing API and if such a change would be backward incompatible.
```

Comment:
There doesn't seem to be a "status" column. I think the last sentence can be removed?